go/cmd/server/logging/logging.go

```diff
@@ -0,0 +1,391 @@
+package logging
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/apache/arrow/go/v8/arrow"
+	"github.com/apache/arrow/go/v8/arrow/array"
+	"github.com/apache/arrow/go/v8/arrow/memory"
+	"github.com/feast-dev/feast/go/internal/feast"
+	"github.com/feast-dev/feast/go/internal/feast/model"
+	"github.com/feast-dev/feast/go/protos/feast/serving"
+	"github.com/feast-dev/feast/go/protos/feast/types"
+	gotypes "github.com/feast-dev/feast/go/types"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const DEFAULT_LOG_FLUSH_INTERVAL = 100 * time.Millisecond
+const DEFAULT_LOG_INSERT_TIMEOUT = 20 * time.Millisecond
+
+type Log struct {
+	// Example: val{int64_val: 5017}, val{int64_val: 1003}
+	EntityValue []*types.Value
+
+	FeatureValues   []*types.Value
+	FeatureStatuses []serving.FieldStatus
+	EventTimestamps []*timestamppb.Timestamp
+	RequestContext  map[string]*types.Value
+	RequestId       string
+}
+
+type MemoryBuffer struct {
+	featureService *model.FeatureService
+	logs           []*Log
+}
+
+type LoggingService struct {
+	memoryBuffer      *MemoryBuffer
+	logChannel        chan *Log
+	fs                *feast.FeatureStore
+	offlineLogStorage OfflineLogStorage
+	logInsertTTl      time.Duration
+	logFlushInterval  time.Duration
+}
+
+func NewLoggingService(fs *feast.FeatureStore, logChannelCapacity int, featureServiceName string, enableLogProcessing bool) (*LoggingService, error) {
+	var featureService *model.FeatureService = nil
+	var err error
+	if fs != nil {
```

Comment:
shouldn't `fs` always be not nil?

Reply:
In some tests it is nil because for example for logger tests, there isn't a need to instantiate a full feature service if I'm only testing individual components.